### PR TITLE
[SYCL][ESIMD] Print performance data for histogram tests in uniform way

### PR DESCRIPTION
**Pull request by sndmitriev**
Signed-off-by: Sergey Dmitriev <serguei.n.dmitriev@intel.com>

### CODE REVIEW

**Review by wenjuan1-intel — DISMISSED** [review]
LGTM

File: SYCL/ESIMD/histogram.cpp [hunk 1]
```diff
@@ -224,8 +231,13 @@ int main(int argc, char *argv[]) {
...
 
-  float kernel_time = kernel_times / num_iters;
-  std::cerr << "GPU kernel time = " << kernel_time << " msec\n";
+  // End timer.
+  double end = timer.Elapsed();
+
+  esimd_test::display_timing_stats(kernel_times, num_iters,
+                                   (end - start) * 1000);
+  // float kernel_time = kernel_times / num_iters;
+  // std::cerr << "GPU kernel time = " << kernel_time << " msec\n";
```
Comment (v-klochkov) [inline SYCL/ESIMD/histogram.cpp]:
```suggestion
```

File: SYCL/ESIMD/histogram_256_slm.cpp [hunk 2]
```diff
@@ -174,14 +178,21 @@ int main() {
...
 
-  float kernel_time = kernel_times / num_iters;
-  std::cerr << "GPU kernel time = " << kernel_time << " msec\n";
+  // End timer.
+  double end = timer.Elapsed();
+
+  esimd_test::display_timing_stats(kernel_times, num_iters,
+                                   (end - start) * 1000);
+  // float kernel_time = kernel_times / num_iters;
+  // std::cerr << "GPU kernel time = " << kernel_time << " msec\n";
```
Comment (v-klochkov) [inline SYCL/ESIMD/histogram_256_slm.cpp]:
```suggestion
```

**Review by v-klochkov — DISMISSED** [review]

File: SYCL/ESIMD/histogram_256_slm.cpp [hunk 3]
```diff
@@ -174,14 +178,21 @@ int main() {
       etime = esimd_test::report_time("kernel time", e, e);
       if (iter > 0)
         kernel_times += etime;
+      else
+        start = timer.Elapsed();
     }
   } catch (cl::sycl::exception const &e) {
     std::cout << "SYCL exception caught: " << e.what() << '\n';
     return e.get_cl_code();
```
Comment (v-klochkov) [inline SYCL/ESIMD/histogram_256_slm.cpp]: Could you please remove such calls as this one. I tis obsolete CL API, which causes warnings
```suggestion
    return 1;
```

**Review by v-klochkov — DISMISSED** [review]

File: SYCL/ESIMD/histogram_256_slm.cpp (hunk 3 above)
Comment (v-klochkov) [reply SYCL/ESIMD/histogram_256_slm.cpp]: BTW, I forgot about this comment when approved 7f6ab8de10f0596665011f41af34c2c5e78e2d94
Would you please also remove get_cl_code() call as well?

Comment (sndmitriev) [reply SYCL/ESIMD/histogram_256_slm.cpp]: Should I change it to e.code().value() instead?

File: SYCL/ESIMD/histogram.cpp [hunk 4]
```diff
@@ -215,17 +220,22 @@ int main(int argc, char *argv[]) {
...
+      else
+        start = timer.Elapsed();
     }
     // SYCL will enqueue and run the kernel. Recall that the buffer's data is
     // given back to the host at the end of scope.
     // make sure data is given back to the host at the end of this scope
   } catch (cl::sycl::exception const &e) {
     std::cout << "SYCL exception caught: " << e.what() << '\n';
-    return e.get_cl_code();
+    return e.code().value();
```
Comment (sndmitriev) [inline SYCL/ESIMD/histogram.cpp]:
```suggestion
    return 1;
```

File: SYCL/ESIMD/histogram_256_slm.cpp [hunk 5]
```diff
@@ -174,14 +178,19 @@ int main() {
       etime = esimd_test::report_time("kernel time", e, e);
       if (iter > 0)
         kernel_times += etime;
+      else
+        start = timer.Elapsed();
     }
   } catch (cl::sycl::exception const &e) {
     std::cout << "SYCL exception caught: " << e.what() << '\n';
-    return e.get_cl_code();
+    return e.code().value();
```
Comment (sndmitriev) [inline SYCL/ESIMD/histogram_256_slm.cpp]:
```suggestion
    return 1;
```

File: SYCL/ESIMD/histogram_256_slm_spec_2020.cpp [hunk 6]
```diff
@@ -160,23 +160,46 @@ int main(int argc, char **argv) {
...
+      etime = esimd_test::report_time("kernel time", e, e);
+      if (iter > 0)
+        kernel_times += etime;
+      else
+        start = timer.Elapsed();
+    }
   } catch (cl::sycl::exception const &e) {
     std::cout << "SYCL exception caught: " << e.what() << '\n';
-    return e.get_cl_code();
+    return e.code().value();
```
Comment (sndmitriev) [inline SYCL/ESIMD/histogram_256_slm_spec_2020.cpp]:
```suggestion
    return 1;
```

File: SYCL/ESIMD/histogram_2d.cpp [hunk 7]
```diff
@@ -135,84 +135,109 @@ int main(int argc, char *argv[]) {
...
+        start = timer.Elapsed();
+    }
 
     // SYCL will enqueue and run the kernel. Recall that the buffer's data is
     // given back to the host at the end of scope.
     // make sure data is given back to the host at the end of this scope
   } catch (cl::sycl::exception const &e) {
     std::cout << "SYCL exception caught: " << e.what() << '\n';
-    return e.get_cl_code();
+    return e.code().value();
```
Comment (sndmitriev) [inline SYCL/ESIMD/histogram_2d.cpp]:
```suggestion
    return 1;
```

File: SYCL/ESIMD/histogram_raw_send.cpp [hunk 8]
```diff
@@ -159,85 +159,110 @@ int main(int argc, char *argv[]) {
...
+        start = timer.Elapsed();
+    }
 
     // SYCL will enqueue and run the kernel. Recall that the buffer's data is
     // given back to the host at the end of scope.
     // make sure data is given back to the host at the end of this scope
   } catch (cl::sycl::exception const &e) {
     std::cout << "SYCL exception caught: " << e.what() << '\n';
-    return e.get_cl_code();
+    return e.code().value();
```
Comment (sndmitriev) [inline SYCL/ESIMD/histogram_raw_send.cpp]:
```suggestion
    return 1;
```

File: SYCL/ESIMD/histogram_256_slm.cpp (hunk 3 above)
Comment (sndmitriev) [reply SYCL/ESIMD/histogram_256_slm.cpp]: I have changed it as you suggested.

**Review by v-klochkov — DISMISSED** [review]

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**